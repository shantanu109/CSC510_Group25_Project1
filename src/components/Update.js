import React, { Component } from 'react';
import { connect } from 'react-redux';

import { searchUsers } from '../actions/search';
import {editItem} from '../actions/auth';
import {clearsearchstate} from '../actions/search';

import 'react-datepicker/dist/react-datepicker.css';

import { fetchJobs } from '../actions/job';
import Job from './Job';


class Update extends Component {
    constructor(props) {
        super(props);
    
        this.state = {
          itemnam:'',
          restid: '',
          itemname:'',
          quantity:'0',
          costperitem:'',
          datebought:'',
          dateexpired:'',
          editMode: false,
        };
      }
    

    handleSearch = (e) => {
        const searchText = e.target.value;
        console.log(searchText)
         
        this.props.dispatch(searchUsers(searchText));
        
      };

   

    clearSearch = () => {
        this.props.dispatch(clearsearchstate([]))
        console.log("UNMOUNT")

    }

    handleInputChange = (fieldName,val) => {

        this.setState({
            [fieldName]: val
        })

      }

    
  handleSave = () => {

    const {restname,restid,itemname,quantity,costperitem,datebought,dateexpired} = this.state;

    const {user} = this.props.auth;
    const {job} = this.props;
    console.log(itemname)

    // this.setState({
    //   restname: user.restname,
    //   restid:user._id
    // })

    this.props.dispatch(editItem(itemname,quantity))

    this.setState({
      itemname:''
    })

   
    
  }

  componentDidMount() {
    this.props.dispatch(fetchJobs());
  }


    
    
    render() {
       
      const {error} = this.props.auth;
      const {user} = this.props.auth;
      const {job} = this.props;
        
      
        return (
            <div>
             

        <div style={{width:'650px',height:'fit-content',marginLeft:'23%'}}>
        {job.map((job) => (
          <>
            <Job job={job} />
          </>
        ))}
        </div>
        
        
        </div>
        
           
        );
    }
}



function mapStateToProps(state) {
    return {
      auth: state.auth,
      results: state.search.results,
      job:state.job,
    };
  }
  
  export default connect(mapStateToProps)(Update);