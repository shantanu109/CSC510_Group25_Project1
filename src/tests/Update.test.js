import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import Update from '../components/Update';
import { configureStore } from '../store';

const store = configureStore();

const dummyVal = {
  auth: {
    user: {
      email: 'test@gmail.com',
      _id: 'tester123',
      name: 'Tester',
    },
  },
  job: [
    {
      _id: 1,
      itemname: 'Test Number 1',
      quantity: 5,
    },
    {
      _id: 2,
      itemname: 'Test Number 2',
      quantity: 10,
    },
  ],
};


test('render', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Update dummyVal={dummyVal} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
