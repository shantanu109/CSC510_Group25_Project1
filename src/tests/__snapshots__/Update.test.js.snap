// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(Update)
    dummyVal={
      Object {
        "auth": Object {
          "user": Object {
            "_id": "tester123",
            "email": "test@gmail.com",
            "name": "Tester",
          },
        },
        "job": Array [
          Object {
            "_id": 1,
            "itemname": "Test Number 1",
            "quantity": 5,
          },
          Object {
            "_id": 2,
            "itemname": "Test Number 2",
            "quantity": 10,
          },
        ],
      }
    }
  />
</ContextProvider>
`;
